haystack/preview/embedding_backends/sentence_transformers_backend.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Optional, Union, Dict
+import hashlib
+import numpy as np
+
+from haystack.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as sentence_transformers_import:
+    from sentence_transformers import SentenceTransformer
+
+
+class SentenceTransformersEmbeddingBackend:
+    """
+    Singleton class to manage SentenceTransformers embeddings.
+    """
+
+    _instances: Dict[str, "SentenceTransformersEmbeddingBackend"] = {}
+
+    def __new__(cls, *args, **kwargs):
+        args_kwargs_str = str(args) + str(kwargs)
```

Comment:
Is it reliable to use a string representation of `args` and `kwargs` to tell if an instance is different? 🤔 
I guess most of the time it will work, but what if one of the args has a string representation that is identical to a different object?
Not sure to be fair, just thinking about out loud.

Reply:
I'm reworking this part to make instance identification more explicit...